Corona-Warn-App/src/main/java/de/rki/coronawarnapp/dccreissuance/ui/consent/DccReissuanceConsentFragment.kt

```diff
@@ -1,13 +1,71 @@
 package de.rki.coronawarnapp.dccreissuance.ui.consent
 
+import android.os.Bundle
+import android.view.View
 import androidx.fragment.app.Fragment
+import androidx.navigation.fragment.findNavController
 import de.rki.coronawarnapp.R
+import de.rki.coronawarnapp.databinding.FragmentDccReissuanceConsentBinding
+import de.rki.coronawarnapp.util.di.AutoInject
+import de.rki.coronawarnapp.util.ui.observe2
+import de.rki.coronawarnapp.util.ui.popBackStack
+import de.rki.coronawarnapp.util.ui.viewBinding
 import de.rki.coronawarnapp.util.viewmodel.CWAViewModelFactoryProvider
 import de.rki.coronawarnapp.util.viewmodel.cwaViewModels
+import setTextWithUrl
 import javax.inject.Inject
 
-class DccReissuanceConsentFragment : Fragment(R.layout.fragment_dcc_reissuance_consent) {
+class DccReissuanceConsentFragment : Fragment(R.layout.fragment_dcc_reissuance_consent), AutoInject {
 
     @Inject lateinit var viewModelFactory: CWAViewModelFactoryProvider.Factory
     private val viewModel: DccReissuanceConsentViewModel by cwaViewModels { viewModelFactory }
+    private val binding: FragmentDccReissuanceConsentBinding by viewBinding()
+
+    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
+        super.onViewCreated(view, savedInstanceState)
+
+        binding.apply {
+            toolbar.setNavigationOnClickListener { popBackStack() }
+            cancelButton.setOnClickListener { popBackStack() }
+            privacyInformation.setOnClickListener { findNavController().navigate(R.id.informationPrivacyFragment) }
+            agreeButton.setOnClickListener { viewModel.startReissuance() }
+
+            viewModel.apply {
+                stateLiveData.observe2(this@DccReissuanceConsentFragment) {
+                    binding.dccReissuanceTitle.text = it.title
+                    binding.dccReissuanceSubtitle.text = it.subtitle
+                    binding.dccReissuanceContent.text = it.content
+                    it.url?.let { url ->
+                        binding.dccReissuanceLink.visibility = View.VISIBLE
```

Comment:
to handle all cases , what about `dccReissuanceLink.isVisible = !it.url.isNullOrEmpty()`